### DISCUSSION TITLE
Can the callbacks generated by subscribing to notifications last forever

### DISCUSSION THREAD

**Discussion by Luo-XiuFeng** [hbldh/bleak#658] [Q&A]
`            while(True):
                await self.client.start_notify(self.noticeuuid, notification_handler())`

1. When I find the uuid of the notification and enable notification, I can get a notification_handler to process, but in order to keep getting the message, I have to use `while(true)` to always enable notification. Is there a better way?
2. There is a serious time problem with `while(true)`. My Bluetooth is sent once a second, but when I print the time in the notification_handler callback, I find that notice is not called strictly in 1 second? How to solve it?

**Comment by dlech** [comment]
What event should trigger the program to end? For example, maybe the program should end when the device disconnects. The you can use `asyncio.Event` to wait for that event. See [examples/disconnect_callback.py](https://github.com/hbldh/bleak/blob/develop/examples/disconnect_callback.py).

**Comment by dlech** [comment]
`start_notify()` should only be called once when the device is connected. Notifications will stay enabled until `stop_notify()` is called or the device disconnects.

**Comment by Luo-XiuFeng** [comment]
> start_notify() should only be called once when the device is connected. Notifications will stay enabled until stop_notify() is called or the device disconnects.

I want to always subscribe to notifications and be able to always receive callbacks to process my data, but after subscribing only through await client.start_notify(CHARACTERISTIC_UUID, notification_handler), I can only receive notifications once and never receive them again. I didn’t call start_stop, nor did I call it unless the device was disconnected。
Suppose I did not comment out the while (true), I will always receive the callback, but the number of callbacks does not match the sending frequency of the Bluetooth data, the callback is very slow, when I comment out the while (true), I will only receive it once Or a second callback.
```python
    async def run(self,debug=0):
        
        log = logging.getLogger(__name__)
        if debug:   
            import sys

        async with BleakClient(self.bleMac) as self.client:
            x = await self.client.is_connected()
            keyuu = '6e'
            for service in self.client.services:
                print(service.uuid)
                if keyuu not in service.uuid:
                    continue

                for char in service.characteristics:
                    print(char.properties)
                    if 'notify' not in char.properties:
                        continue
                    else:
                        print(char.uuid)
                        self.noticeuuid = char.uuid
                        try:
                            value = bytes(await self.client.read_gatt_char(char.uuid))
                        except Exception as e:
                            value = str(e).encode()
                    

                    for descriptor in char.descriptors:
                        value = await self.client.read_gatt_descriptor(descriptor.handle)
                        # await asyncio.sleep(1)
                        #await self.client.stop_notify(CHARACTERISTIC_UUID)

            # while(True):
                await self.client.start_notify(self.noticeuuid,self.notification_handler)
```

**Comment by Luo-XiuFeng** [comment]
> `start_notify()` should only be called once when the device is connected. Notifications will stay enabled until `stop_notify()` is called or the device disconnects.

Hello, what should I do to keep being notified at all times? Or is there an example for reference?
